python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,576 @@
+            help="Table to ingest into")
+        parser.add_argument(
+            "--view", dest="viewName",
+            help="View to create containing column aliases")
+        parser.add_id_argument(
+            "--id", pipeBase.DatasetArgument("dstype"),
+            help="Dataset data id to ingest")
+            # Use DatasetArgument to require dataset type be specified on
+            # the command line
+        return parser
+
+    def runFile(self, fileName, tableName, host, db, port=3306, user=None, viewName=None):
+        """Ingest a BaseCatalog specified by a filename.
+        """
+        cat = afwTable.BaseCatalog.readFits(fileName)
+        self.ingest(cat, tableName, host, db, port, user, viewName)
+
+    def run(self, dataRef, dstype, tableName, host, db, port=3306, user=None, viewName=None):
+        """Ingest a BaseCatalog specified by a dataref and dataset type.
+        """
+        self.ingest(dataRef.get(dstype), tableName, host, db, port, user, viewName)
+
+    @pipeBase.timeMethod
+    def ingest(self, cat, tableName, host, db, port=3306, user=None, viewName=None):
+        """Ingest a BaseCatalog passed as an object.
+
+        @param cat (BaseCatalog) Catalog to ingest.
+        @param tableName (str)   Name of the database table to create.
+        @param host (str)        Name of the database host machine.
+        @param db (str)          Name of the database to ingest into.
+        @param port (int)        Port number on the database host.
+        @param user (str)        User name to use for the database.
+        @param viewName (str)    Name of the database view to create.
+        """
+        tableName = quoteIdentifier(tableName)
+        viewName = quoteIdentifier(viewName) if viewName else None
+        with closing(self.connect(host, port, db, user)) as conn:
+            # Determine the maximum query length (MySQL-specific) if not
+            # configured.
+            if self.config.maxQueryLen is None:
+                with closing(conn.cursor()) as cursor:
+                    cursor.execute(
+                        """SELECT variable_value
+                        FROM information_schema.session_variables
+                        WHERE variable_name = 'max_allowed_packet'
+                        """
+                    )
+                    maxQueryLen = int(cursor.fetchone()[0])
+            else:
+                maxQueryLen = self.config.maxQueryLen
+            self.log.logdebug("maxQueryLen: {}".format(maxQueryLen))
+            self._createTable(conn, tableName, cat.schema)
+            if viewName is not None:
+                self._createView(conn, tableName, viewName, cat.schema)
+            self._ingest(conn, cat, tableName, maxQueryLen)
+
+    @staticmethod
+    def connect(host, port, db, user=None):
+        # Map BIT columns to Python bools
+        conv = dict(MySQLdb.converters.conversions)
+        conv[MySQLdb.constants.FIELD_TYPE.BIT] = lambda v: v != '\x00'
```

Comment:
The default conversion is weird (BIT -> '\x00' or '\x01'). I actually only really need the BIT -> bool mapping in the unit test, as the ingest code doesn't read from the database at all. I'll move this logic (or some equivalent) into the unit test with a nicer comment.